### PR TITLE
Update Storybook to v6

### PR DESCRIPTION
**Pull request by koba04**
## Overview

This is based on [the migration guilde](https://medium.com/storybookjs/storybook-6-migration-guide-200346241bb5).
Thie PR focuses on updating the storybook version, so I didn't update to use new features introduced in v6.
I'll take this as other PRs.

## What I did

- `npx sb@next upgrade`
- add `react` and `react-dom` into the resolutions field (https://github.com/storybookjs/storybook/issues/11904)
- remove `webpack.config.js` for TypeScript
- migrate settings for storybook

### CODE REVIEW

**Comment by koba04** [comment]
[The diff ](https://smarthr-ui-snapshots.s3.amazonaws.com/566da28c6a97900a5033aaa9d954cf66aed99f90/index.html?stat=q1ZKS8zMSU3xLEnNLXbOL80rUbIy0FHKSy1HFjHWUUpJzUktQVUHFC1ILC5GFTQzrQUA#deleted-Dialog/opened/MessageDialog.png) is caused by https://github.com/kufu/smarthr-ui/pull/952/commits/693f4174259fa037fb29a670d2df25e897a0a01f, so it seems to be fine because the diffs are caused by renaming files.
But I prefer to remove `preview-head.html` rather than keeping the styles using `!important`.

**Review by nabeliwo — COMMENTED** [review]
I encountered the following problem. Do you understand the cause?

- When I run `npm storybook` locally, panel such as addon disappear
    - image
- I get frequent `not found` warnings
    - image

**Comment by koba04** [comment]
@nabeliwo 

> When I run npm storybook locally, panel such as addon disappear

Hmm, my environment works fine, so could you remove your sessionStorage values?

image

> I get frequent not found warnings

Thanks! That is caused by changing the way compiling TypeScript from `ts-loader` to `babel-loader`.
I've fixed the warnings with `export type`.

**Comment by wmoai** [comment]
@koba04 Thanks!
I fixed the problem of stylelint in #1034, so I think the CI failures will go away by getting master in.

File: .storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      }
+    },
+    {
+      name: '@storybook/addon-storysource',
+      options: {
+        parser: 'typescript'
+      }
+    },
+    '@storybook/addon-a11y',
+    'storybook-readme'
```
Comment (wmoai) [inline .storybook/main.js:19]: `storybook-readme` should move to the top of addons array in order to show README tab first.
#1041

**Review by wmoai — APPROVED** [review]
LGTM!

**Review by nabeliwo — APPROVED** [review]

**Pull request merged**